test/ffi/booster.jl

```diff
@@ -286,27 +290,50 @@ end
 
 @testset "LGBM_BoosterSaveModel" begin
 
-    # Needs implementing
-    @test_broken false
+    booster = LightGBM.LGBM_BoosterCreateFromModelfile(joinpath(@__DIR__, "data", "test_tree"))
+    model_path = joinpath(@__DIR__, "data", "test_model.txt")
+    LightGBM.LGBM_BoosterSaveModel(booster, 0, 0, 0, model_path )
+    @test isfile(model_path)
+    rm(model_path; force = true)
 
 end
 
 
 @testset "LGBM_BoosterSaveModelToString" begin
 
     booster = LightGBM.LGBM_BoosterCreateFromModelfile(joinpath(@__DIR__, "data", "test_tree"))
-    string_repr = LightGBM.LGBM_BoosterSaveModelToString(booster, 0, 0)
+    string_repr = LightGBM.LGBM_BoosterSaveModelToString(booster, 0, 0, 0)
     # so it turns out that the string save and file save aren't necesarily the same so..
     # check a bunch of expected substrings, etc
     @test occursin("version=v3", string_repr)
     @test occursin("num_leaves=1", string_repr)
     @test occursin("end of trees", string_repr)
-    @test occursin("feature importances:", string_repr)
+    @test occursin("feature_importances:", string_repr)
     @test occursin("parameters:", string_repr)
     @test occursin("[convert_model: gbdt_prediction.cpp]", string_repr)
     @test occursin("Tree=0", string_repr)
     @test occursin("end of parameters", string_repr)
 
+    # this is an additional test to check the presence and corectness of split and gain after saving model to string
+    # due to LGBM v3.0.0 changing the LGBM_BoosterSaveModelToString API that includes now either split or gain parameters
+    # this test generates a model, gets the split and gain,
+    # saves model to string using case with split and gain parameters accordingly with the updated API call
+    # loads the models saved with these params and checks the split and gain values are as expected
+    booster = LightGBM.LGBM_BoosterCreateFromModelfile(joinpath(@__DIR__, "data", "gain_test_booster"))
+    split_importance = LightGBM.LGBM_BoosterFeatureImportance(booster, 0, 0)
+    gain_importance = LightGBM.LGBM_BoosterFeatureImportance(booster, 0, 1)
+    split_save = LightGBM.LGBM_BoosterSaveModelToString(booster, 0, 0, 0)
+    gain_save = LightGBM.LGBM_BoosterSaveModelToString(booster, 0, 0, 1)
+    split_load = LightGBM.LGBM_BoosterLoadModelFromString(split_save)
+    gain_load = LightGBM.LGBM_BoosterLoadModelFromString(gain_save)
+    split_importance_loaded = LightGBM.LGBM_BoosterFeatureImportance(split_load, 0, 0)
+    gain_importance_loaded = LightGBM.LGBM_BoosterFeatureImportance(gain_load, 0, 1)
+    expected_gain = [89.73980999, 65.49232054, 112.80447054, 107.81817985, 124.81229973]
```

Comment:
Why have you coded these values in here? The way I would imagine this test to work is as follows:

* load that model above (or any other)
* compute both its gain and split importance
* save the model twice, once with each of those gain and split parameters
* load each of the saved models to a new model
* for EACH model, compute gain and split importances, and show that both values are the same as the originals for both models

This then shows that we don't have to concern ourselves with which value we pass by default when using save model to string internally for serialisation/deserialisation purposes.

Reply:
hopefully now it's better